### DISCUSSION TITLE
Add Opsgenie Integration to Alerts Modal

### DISCUSSION THREAD

**Discussion by KentonParton** [apache/superset#18450] [Ideas]
**Is your feature request related to a problem? Please describe.**
Superset currently doesn't support Opsgenie as an alert integration which many companies use.

**Describe the solution you'd like**
I would like to add Opsgenie as a notification method to the alert modal using the [Opsgenie API integration](https://docs.opsgenie.com/docs/alert-api#create-alert). The easiest implementation will be to add Opsgenie to the list of notification methods and allow users to create a custom payload as shown in the Opsgenie API docs.

Alternatively, UI components can be created for the different payload fields such as priority, responders, tags etc.

When Opsgenie is selected as a notification method, "Message Content" should be greyed out.

**Describe alternatives you've considered**
Currently, alerts can be dispatched to Opsgenie via Slack; however, this creates a dependency on Slack and is not ideal.

**Additional context**
<img width="1430" alt="Screen Shot 2021-02-19 at 2 17 17 PM" src="https://user-images.githubusercontent.com/20202312/108503555-3ea02480-72bd-11eb-891f-0fa38c6385eb.png">

**Comment by willbarrett** [comment]
Hey @KentonParton - I think we'd like to eventually move to a plugin architecture that would allow additional notification types to be easily added by the community. Is that an effort you'd be interested in participating in?

**Comment by KentonParton** [comment]
Hey @willbarrett thanks for the comment. While I'd love to try the plugin approach, I can't say I am hugely familiar with Superset. Would it be okay to keep the scope to adding the Opsgenie implementation then have a follow-up PR trying the plugin route?

**Comment by willbarrett** [comment]
@KentonParton yes, that would be fine for now. Go for it!

**Comment by zuzana-vej** [comment]
Overall Email Alerting feature enhancements here: https://github.com/apache/superset/issues/13856 

@KentonParton  are you still planning to tackle this / otherwise please feel free to close the ticket.
